source/tyr/tyr/resources.py

```diff
@@ -639,6 +648,12 @@ def map_args_to_instance(attr_name):
             new = copy.deepcopy(instance.max_nb_crowfly_by_mode)
             new.update(max_nb_crowfly_by_mode)
             instance.max_nb_crowfly_by_mode = new
+
+            for provider_id in args.equipment_details_providers:
+                equipment_provider = models.EquipmentsProvider.query.get(provider_id)
+                if equipment_provider:
+                    instance.equipment_details_providers.append(equipment_provider)
```

Comment:
you probably need to clear the list before.
And I didn't see any test on that

Reply:
Done :heavy_check_mark: